archive/qna_engine.py

```
@@ -255,3 +255,70 @@ def get_youtube_transcript(url: str) -> str:
+        prompt_template += "\nThe response should be in JSON format. \n {format_instructions}"
+
+        MCQ_prompt = PromptTemplate(
+            input_variables=["num", "topic"],
+            template=prompt_template,
+            partial_variables={"format_instructions": format_instructions}
+        )
+
+        if llm:
+            llm = llm
+        else:
+            llm = self.llm  # Use the initialized LLM from the class
```

Ruff

Use ternary operator `llm = llm if llm else self.llm` instead of `if`-`else`-block

(SIM108)